pkg/conference/processor.go

```diff
@@ -0,0 +1,220 @@
+		participant.logger.Info("Started renegotiation")
+		participant.sendDataChannelMessage(event.SFUMessage{
+			Op:       event.SFUOperationOffer,
+			SDP:      msg.Offer.SDP,
+			Metadata: c.getAvailableStreamsFor(participant.id),
+		})
+
+	case peer.DataChannelMessage:
+		participant.logger.Debug("Received data channel message")
+		var sfuMessage event.SFUMessage
+		if err := json.Unmarshal([]byte(msg.Message), &sfuMessage); err != nil {
+			c.logger.Errorf("Failed to unmarshal SFU message: %v", err)
+			return
+		}
+
+		c.handleDataChannelMessage(participant, sfuMessage)
+
+	case peer.DataChannelAvailable:
+		participant.logger.Info("Connected data channel")
+		participant.sendDataChannelMessage(event.SFUMessage{
+			Op:       event.SFUOperationMetadata,
+			Metadata: c.getAvailableStreamsFor(participant.id),
+		})
+
+	default:
+		c.logger.Errorf("Unknown message type: %T", msg)
+	}
+}
+
+// Handle the `SFUMessage` event from the DataChannel message.
+func (c *Conference) handleDataChannelMessage(participant *Participant, sfuMessage event.SFUMessage) {
+	switch sfuMessage.Op {
+	case event.SFUOperationSelect:
+		participant.logger.Info("Received select request over DC")
+
+		// Find tracks based on what we were asked for.
+		tracks := c.getTracks(sfuMessage.Start)
+
+		// Let's check if we have all the tracks that we were asked for are there.
+		// If not, we will list which are not available (later on we must inform participant
+		// about it unless the participant retries it).
+		if len(tracks) != len(sfuMessage.Start) {
+			for _, expected := range sfuMessage.Start {
+				found := slices.IndexFunc(tracks, func(track *webrtc.TrackLocalStaticRTP) bool {
+					return track.StreamID() == expected.StreamID && track.ID() == expected.TrackID
+				})
+
+				if found == -1 {
+					c.logger.Warnf("Track not found: %s", expected.TrackID)
+				}
+			}
+		}
+
+		// Subscribe to the found tracks.
+		for _, track := range tracks {
+			if err := participant.peer.SubscribeTo(track); err != nil {
+				participant.logger.Errorf("Failed to subscribe to track: %v", err)
+				return
+			}
+		}
+
+	case event.SFUOperationAnswer:
+		participant.logger.Info("Received SDP answer over DC")
+
+		if err := participant.peer.ProcessSDPAnswer(sfuMessage.SDP); err != nil {
+			participant.logger.Errorf("Failed to set SDP answer: %v", err)
+			return
+		}
+
+	case event.SFUOperationPublish:
+		participant.logger.Info("Received SDP offer over DC")
+
+		answer, err := participant.peer.ProcessSDPOffer(sfuMessage.SDP)
+		if err != nil {
+			participant.logger.Errorf("Failed to set SDP offer: %v", err)
+			return
+		}
+
+		participant.streamMetadata = sfuMessage.Metadata
+
+		participant.sendDataChannelMessage(event.SFUMessage{
+			Op:       event.SFUOperationAnswer,
+			SDP:      answer.SDP,
+			Metadata: c.getAvailableStreamsFor(participant.id),
+		})
+
+	case event.SFUOperationUnpublish:
+		participant.logger.Info("Received unpublish over DC")
```

Comment:
We should be resending the metadata here and removing the track, right?

Reply:
As discussed in our DM, we only need an `Offer` event and can get rid of `Publish` and `Unpublish` in the future